docs/developer/index.rst

```diff
@@ -724,13 +769,135 @@ current way that models are checked:
   ``ndim`` value (ignoring those models whose dimensionality
   is set to ``None``). If there is a mis-match then a
   :py:class:`sherpa.utils.err.ModelErr` is raised.
+* as :ref:`descibed below <data_design_ndim>`, the dimensions of
+  data and model can be compared.
 
-An alternative approach would have been to introdude 1D and 2D
+An alternative approach would have been to introduce 1D and 2D
 specific classes, from which all models derive, and then require the
 parent classes to match. This was not attempted as it would require
 significantly-larger changes to Sherpa (but this change could still be
 made in the future).
 
+.. _data_design:
+
+The data class
+--------------
+
+Prior to Sherpa 4.14.1, the `~sherpa.data.Data` object did not have
+many explicit checks on the data it was sent, instead relying on
+checks when the data was used. This has been changed to apply
+validation checks when fields are changed, rather than when the data
```

Comment:
Just avoiding to start several sentences in a raw with "This has been"
```suggestion
checks when the data was used. Now, validation checks  are done
when fields are changed, rather than when the data
```